src/bindings/angular/chart/index.js

```diff
@@ -58,6 +58,10 @@ export class ChartDirective {
               $scope.$applyAsync();
               $scope.$emit('babbage-ui.ready', component, data, error);
             });
+            component.on('click', function(component, item) {
+              $scope.$emit('babbage-ui.click', component, item);
+              $scope.$applyAsync();
```

Comment:
Why is the order from this event `$emit` and then `$applySync`, while the event above do the inverse (`$applySync` and then `$emit`)?

Reply:
I'll admit, I don't know. I copied it from the existing pie chart bindings: https://github.com/openspending/babbage.ui/blob/2905409f26108ff859f925e5e2481fce8acd0b8e/src/bindings/angular/pie/index.js#L53-L65

Reply:
From what I understand, the `$applyAsync` is required because we're modifying the `$scope` outside of an Angular callback. However, as on click we only have `$emit`, I'm not sure if these are necessary.

I don't know, so happy to go with whatever you decide.